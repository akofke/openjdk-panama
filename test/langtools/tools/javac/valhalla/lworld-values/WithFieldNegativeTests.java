/*
 * @test /nodynamiccopyright/
 * @summary Check various withfield constrains
 *
 * @compile/fail/ref=WithFieldNegativeTests.out -XDrawDiagnostics WithFieldNegativeTests.java
 */

__ByValue final class A {
    final int x = 10;
    static final int sx = 10;

    __ByValue final class B {

        final A a = __MakeDefault A();

        void foo(A a) {
            a.x = 100; // OK, same nest.
            a.sx = 100; // Error.
        }
    }

    void withfield(B b) {
            b.a.x = 11; // Error, at least for now.
    }

    void foo(A a, final A fa) {
        a.x = 100; // OK.
        (a).x = 100; // OK.
        fa.x = 100; // Error.
        x = 100;  // Error, this is const.
        this.x = 100; // Error.
        A.this.x = 100; // Error.
    }
}

class C {
    void foo(A a) {
        a.x = 100; // Not OK.
        a.sx = 100; // Error
    }
}
