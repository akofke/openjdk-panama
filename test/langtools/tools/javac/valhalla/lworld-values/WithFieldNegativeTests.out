WithFieldNegativeTests.java:18:14: compiler.err.cant.assign.val.to.final.var: sx
WithFieldNegativeTests.java:23:16: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:29:11: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:30:9: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:31:13: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:32:15: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:38:10: compiler.err.cant.assign.val.to.final.var: x
WithFieldNegativeTests.java:39:10: compiler.err.cant.assign.val.to.final.var: sx
8 errors
